added schema for data modeling blog (#23)
vendia/examples: share/data-modeling/domain-driven-design/pattern1/README.md

# Pattern 1: Single Domain, Single Bounded Context, and a Single Uni

# Scenario

> The customer Relationship Management(CRM) Team wants to build a single source of truth for customer account data which includes first name, last name, customerId, personal details, contact details, and address. The CRM team will frequently query customer data using customerId. 

Reviewer: ```suggestion
The Customer Relationship Management(CRM) Team wants to create a single source of truth for customer account data, which includes first name, last name, customerId, personal details, contact details, and address. The CRM Team will frequently query customer data using customerId. 
```

# Modeling

> Since CRM is an independent function we will assume it is a domain on its own. Inside CRM, we will have multiple sub-domains; customer account is one such sub-domain. You can model a customer account as a Bounded Context.  Since CRM will perform frequent lookup, search, and filter on customer records by customerId, you will index customer accounts on customerId property. 

Reviewer: ```suggestion
Since CRM is an independent business unit we will assume it is a domain on its own. Inside CRM, we will have multiple sub-domains; customer account is one such sub-domain. You can model a customer account as a Bounded Context.  Since CRM will perform frequent lookup, search, and filter on customer records by customerId, you will index customer accounts on customerId property. 
```

# Create a Uni

1. Login to Vendia Share [here](https://share.vendia.net/login). 

Don’t have a login, you can sign up for one [here](https://share.vendia.net/). 

2. Create Uni

> Click the 'Create Universal Application' button on the top-right. Next, use the ‘Create your own option. 

Reviewer: ```suggestion
Click the 'Create Universal Application' button on the top-right. Next, use the `Create your own` option. 
```

3. Fill out details in the Uni Creation Wizard. 

> 1. **Step 1:** Give your Uni a name. Since this Uni will be owned and operated by the CRM team. I will give it the name ‘CRM Uni’. Click ‘Next’

Reviewer: We should recommend:

1. They prefix their Uni name with `test-` so its name can be reused
2. They pick a unique Uni name since the `vendia.net` namespace is shared by all Starter and Individual Tier users, which will cause naming collisions

Author: I will also add a suggestion to delete Uni at the end of each pattern as free tier will have license to create a  single Uni

           "type": "array",
           "items": {
               "type": "object",
               "properties": {
		…
```

The nested complex type objects in the ‘CustomerAccount’ such as ‘ContactDetails’.

```
"contactDetails": {
                       "description": "customer contact details",
                       "type": "object",
                       "properties": {
                           "email": {
                               "description": "email Id of a customer",
                               "type": "string"
                           },
                           "phone": {
                               "description": "phone number of a customer",
                               "type": "string"
                           }
                       }
                   }
```

The required constraint on the customerId, firstname, lastName, and on the next complex type address, and property of address i.e zipcode.

```
"required": [
                    "customerId",
                    "firstName",
                    "lastName",
                    "address"
                ]
```

4. Press “Create”. Wait for 5 minutes for Share to finish Uni provisioning.

> After Uni Provisioning is complete. You will see that the Uni with the name you chose will appear in the running status on the home page. 

Reviewer: ```suggestion
After Uni provisioning is complete, you will see that the Uni with the name you chose will appear in the `RUNNING` status on the home page. 
```

# Explore the Uni

> Click the Uni you just created. Click on the ‘Entity Explorer’. You will notice ‘Schema Owner’ and CustomerAccount’ Entity’ on the left. 

Reviewer: ```suggestion
Click the Uni you just created. Click on the `Entity Explorer`. You will notice `Schema Owner` and `CustomerAccount Entity` on the left. 
```

Author: accepted

> Schema Owner is a Value Object; It’s a singleton value object with no unique identifier. You will use it to keep track of who is the Uni ‘owner’ and ‘emailId’ of the group owning it. It is good to have this information in the Uni itself. Click Edit and fill the ‘group’ and ‘emailId’ properties.

Reviewer: ```suggestion
Schema Owner is a Value Object; it’s a singleton value object with no unique identifier. You will use it to keep track of who is the Uni `owner` and `emailId` of the group owning it. It is good to have this information in the Uni itself. Click Edit and fill the `group` and `emailId` properties.
```

> Let’s create a customer record. Click ‘CustomerAccount’ entity. Click ‘Create Customer Account’. Enter customer account details. For CustomerId, pick a number. I use the [10001,10002…] series for testing. In the real world, either you will have your own sequence generator for customerId, or you will use the customerId from enterprise source of truth systems elsewhere, or you can use the id property value automatically created by Uni for a customer record. 

Reviewer: Not going to inline edit all of these.  I believe going for `codesnippets` for the items you want to call out by name is a good practice.  Before you make all those changes, @brian-vendia  - does that match our new conventions?
